var/spack/repos/builtin/packages/python/package.py

```diff
@@ -114,22 +115,28 @@ def install(self, spec, prefix):
                       'user configurations are present.').format(self.version))
 
         # Need this to allow python build to find the Python installation.
-        env['PYTHONHOME'], env['PYTHONPATH'] = prefix, prefix
-        env['MACOSX_DEPLOYMENT_TARGET'] = '10.6'
+        spack_env.set('PYTHONHOME', prefix)
+        spack_env.set('PYTHONPATH', prefix)
+        spack_env.set('MACOSX_DEPLOYMENT_TARGET', platform.mac_ver()[0])
+
+    def configure_args(self):
+        spec = self.spec
 
-        # Rest of install is pretty standard except setup.py needs to
-        # be able to read the CPPFLAGS and LDFLAGS as it scans for the
-        # library and headers to build
+        # setup.py needs to be able to read the CPPFLAGS and LDFLAGS
+        # as it scans for the library and headers to build
         dep_pfxs = [dspec.prefix for dspec in spec.dependencies('link')]
         config_args = [
-            '--prefix={0}'.format(prefix),
             '--with-threads',
-            '--enable-shared',
             'CPPFLAGS=-I{0}'.format(' -I'.join(dp.include for dp in dep_pfxs)),
             'LDFLAGS=-L{0}'.format(' -L'.join(dp.lib for dp in dep_pfxs)),
         ]
-        if spec.satisfies("platform=darwin") and ('%gcc' in spec):
-            config_args.append('--disable-toolbox-glue')
+        if spec.satisfies('platform=darwin'):
+            if '%gcc' in spec:
+                config_args.append('--disable-toolbox-glue')
+        else:
+            # --enable-shared causes problems on macOS and is not recommended:
```

Comment:
i suggest that this comment is rewored, i.e. "may cause problems in some scenarious" or alike because it does work in other cases:
```
$ spack load python
$ which python
/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/python-2.7.13-hcpozzgaht7gddrvyrn2fprcm5tmqqii/bin/python
$ python
Python 2.7.13 (default, Mar  3 2017, 22:36:11)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import io
>>>
```